src/logger/manager.rs

```diff
@@ -1,43 +1,52 @@
-use log::*;
-use std::io::Write;
-
 use crate::cli;
-use env_logger::fmt::{Color, Style, StyledValue};
 
-// Change style based on the message log level
-fn colored_level<'a>(style: &'a mut Style, level: Level) -> StyledValue<'a, &'static str> {
-    match level {
-        Level::Trace => style.set_color(Color::Magenta).value("TRACE"),
-        Level::Debug => style.set_color(Color::Blue).value("DEBUG"),
-        Level::Info => style.set_color(Color::Green).value("INFO "),
-        Level::Warn => style.set_color(Color::Yellow).value("WARN "),
-        Level::Error => style.set_color(Color::Red).value("ERROR"),
-    }
-}
+use tracing::*;
+use tracing_log::LogTracer;
+use tracing_subscriber::{fmt, layer::SubscriberExt, prelude::*, EnvFilter, Layer};
 
 // Start logger, should be done inside main
 pub fn init() {
-    if cli::manager::is_verbose() {
-        std::env::set_var("RUST_LOG", "debug");
-    }
+    // Redirect all logs from libs using "Log"
+    LogTracer::init_with_filter(tracing::log::LevelFilter::Trace).expect("Failed to set logger");
+
+    // Configure the console log
+    let env_filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| {
+        if cli::manager::is_verbose() {
+            EnvFilter::new("debug")
+        } else {
+            EnvFilter::new("info")
+        }
+    });
+    let console_layer = fmt::Layer::new()
+        .with_writer(std::io::stdout.with_max_level(Level::DEBUG))
+        .with_ansi(true)
+        .with_file(true)
+        .with_line_number(true)
+        .with_span_events(fmt::format::FmtSpan::FULL)
+        .with_target(false)
+        .with_thread_ids(true)
+        .with_thread_names(true)
+        .with_filter(env_filter);
+
+    // Configure file log
+    let dir = cli::manager::log_path();
+    let file_appender = tracing_appender::rolling::daily(dir, "mavlink-camera-manager.log");
+    let file_layer = fmt::Layer::new()
+        .with_writer(file_appender)
+        .with_ansi(false)
+        .with_file(true)
+        .with_line_number(true)
+        .with_span_events(fmt::format::FmtSpan::FULL)
+        .with_target(false)
+        .with_thread_ids(true)
+        .with_thread_names(true)
+        .with_filter(EnvFilter::new("debug"));
```

Comment:
`LevelFilter`, this will make more safer since is based in enum and not strings.

Reply:
see https://github.com/bluerobotics/mavlink-camera-manager/pull/121#discussion_r957380663

Reply:
```suggestion
        .with_filter(LevelFilter::DEBUG);
```